### PR TITLE
#17 added functionality of emailSender and new feture [email confirming]

### LINKED ISSUE #17: Register email confirmation

### PR DESCRIPTION
**Pull request by Prestionyk**
Should close #17

### CODE REVIEW

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs [hunk 1]
```diff
@@ -105,7 +114,30 @@ private async Task<Response> RegisterUser(RegisterModel model, UserManager<Ident
...
             return null;
         }
+        public async Task<Response> ConfirmEmail(ConfirmEmailModel model)
+        {
+            var user = await _userManager.FindByNameAsync(model.UserName);
+
+            var result = await _userManager.ConfirmEmailAsync(user, model.Token);
+
+            if (result.Succeeded)
+                return new Response { StatusCode = 200, Message = "Email confirmed succesfully" };
```
Comment (wikhag) [inline WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Use enum for response codes

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs [hunk 2]
```diff
@@ -105,7 +114,30 @@ private async Task<Response> RegisterUser(RegisterModel model, UserManager<Ident
...
+            return new Response { StatusCode = 400, Message = "Email was'nt able to confirm via this link" };
```
Comment (wikhag) [inline WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Use enum for response codes

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs [hunk 3]
```diff
@@ -105,7 +114,30 @@ private async Task<Response> RegisterUser(RegisterModel model, UserManager<Ident
             if (!result.Succeeded)
                 return new Response { StatusCode = 422, Message = "User creation failed! Please check user details and try again." };
```
Comment (wikhag) [inline WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Use enum for response codes

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs [hunk 4]
```diff
@@ -86,7 +95,7 @@ public async Task<Response> RegisterAdmin(RegisterModel model)
                     await _userManager.AddToRoleAsync(user, UserRoles.Admin);
                 }
 
-                return new Response { StatusCode = 201, Message = "User created." };
+                return new Response { StatusCode = 201, Message = "Admin user created. Now confirm your email." };
```
Comment (wikhag) [inline WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Use enum for response codes

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs [hunk 5]
```diff
@@ -58,14 +68,13 @@ public async Task<Response> Login(LoginModel model)
...
         }
-
         public async Task<Response> Register(RegisterModel model)
         {
             Identity.User user = new();
             Response res = await RegisterUser(model, _userManager, user);
 
             if (res == null)
-                return new Response { StatusCode = 201, Message = "User created successfully!" };
+                return new Response { StatusCode = 201, Message = "User created successfully! Now confirm your email." };
```
Comment (wikhag) [inline WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Use enum for response codes

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs [hunk 6]
```diff
@@ -58,14 +68,13 @@ public async Task<Response> Login(LoginModel model)
 
             return new Response { StatusCode = 422, Message = "Username or password is not correct!" };
```
Comment (wikhag) [inline WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Use enum for response codes

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs [hunk 7]
```diff
@@ -16,19 +20,25 @@ public class UserAuthenticationService : IUserAuthenticationService
...
+            _emailService = emailService;
         }
 
         public async Task<Response> Login(LoginModel model)
         {
             var user = await _userManager.FindByNameAsync(model.Username);
             if (user != null && await _userManager.CheckPasswordAsync(user, model.Password))
             {
+                if(!await _userManager.IsEmailConfirmedAsync(user))
+                    return new Response { StatusCode = 422, Message = "Invalid username or password!" };
```
Comment (wikhag) [inline WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Use enum for response codes

**Review by wikhag — CHANGES_REQUESTED** [review]

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs (hunk 1 above)
Comment (Prestionyk) [reply WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Changed in another branch

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs (hunk 2 above)
Comment (Prestionyk) [reply WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Changed in another branch

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs (hunk 4 above)
Comment (Prestionyk) [reply WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Changed in another branch

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs (hunk 5 above)
Comment (Prestionyk) [reply WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Changed in another branch

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs (hunk 7 above)
Comment (Prestionyk) [reply WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Changed in another branch

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs (hunk 3 above)
Comment (Prestionyk) [reply WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Changed in another branch

File: WebAPI/Services/User/Authentication/UserAuthenticationService.cs (hunk 6 above)
Comment (Prestionyk) [reply WebAPI/Services/User/Authentication/UserAuthenticationService.cs]: Changed in another branch

**Review by Justyn-98 — APPROVED** [review]

**Review by wikhag — APPROVED** [review]

**Pull request merged**